### PR TITLE
Support newer versions of `find` that no longer support the "-perm +111" option.

### PR DESCRIPTION
**Pull request by thequux**
Signed-off-by: TQ Hirsch thequux@thequux.com

This patch is necessary at least for Cygwin.

### CODE REVIEW

**Comment by molovo** [comment]
This completely breaks the script on OSX - running `git hooks` results in no hooks being returned at all.

File: git-hooks [hunk 1]
```diff
@@ -51,9 +51,9 @@ function list_hooks_in_dir
     level="${2}"
     find --help 2>&1 | grep -- '-L' 2>/dev/null >/dev/null
     if [ $? -eq 1 ] ; then
-        find "${path}/" -mindepth ${level} -maxdepth ${level} -perm +111 -type f 2>/dev/null | grep -v "^.$" | sort
+        find "${path}/" -mindepth ${level} -maxdepth ${level} -perm /111 -type f 2>/dev/null | grep -v "^.$" | sort
     else
-        find -L "${path}/" -mindepth ${level} -maxdepth ${level} -perm +111 -type f 2>/dev/null | grep -v "^.$" | sort
+        find -L "${path}/" -mindepth ${level} -maxdepth ${level} -perm /111 -type f 2>/dev/null | grep -v "^.$" | sort
```
Comment (thequux) [inline git-hooks]: I'd rather switch on the implementation of find; I'd expect that the BSDs would all require +111, and who knows which platforms use GNU findutils. The other option is to go full POSIX and always use `\( -perm -100 -o -perm -010 -o -perm -001 \)`, which *should* work on any POSIX-compatible find. Do you have a preference?

**Comment by molovo** [comment]
POSIX compliance is always good, so if you can get that working go for it. It might also be worth trying `-perm -a+x`, I think that would give us the same result, and should work on most, if not all platforms.

**Comment by thequux** [comment]
`-perm -a+x` is exactly equivalent to `-perm -111`, which requires that *all* executable bits are set. What we really want is GNU findutils `-executable` option, which finds files that are executable by the current user. Unfortunately, that's not really portable, and I don't know of any other command-line interface to `access(2)` (much less a convenient, portable one)

**Comment by molovo** [comment]
@thequux ah, nice. Your patch looks good in theory, but I'm away from my desk at the moment. I'll try and test it and get it merged tomorrow.

**Pull request merged**